crates/sui-swarm/src/memory/node.rs

```diff
@@ -221,7 +225,7 @@ mod test {
     #[tokio::test]
     async fn start_and_stop() {
         telemetry_subscribers::init_for_testing();
-        let mut swarm = Swarm::builder().build();
+        let mut swarm = Swarm::builder().build(None);
```

Comment:
The purpose of the builder pattern is to avoid optional params and/or too many params. It should look like:

```
Swarm::builder().with_fullnodes(1).build();
```

Reply:
completely agree